arch/src/aarch64/gic/gicv3_its.rs

```diff
@@ -161,7 +159,7 @@ pub mod kvm {
         vcpu_count: u64,
     }
 
-    #[derive(Versionize)]
```

Comment:
Switching away from `Versionize` only makes sense if you never expect the structure to change?

Reply:
The intention of this change is for the consideration of moving `GicV3ItsState` into `hypervisor` crate.
Deriving `Versionize` requires introducing 3 more crates: `versionize`, `versionize_derive` and `vm-migration`. Adding `vm-migration` makes `hypervisor` independent from other local crates.

I looked around and followed the way of "VcpuKvmState". Does "VcpuKvmState" share the same risk for future change?

Reply:
> Switching away from Versionize only makes sense if you never expect the structure to change?

Gicv3ItsState is just describing register state so I doubt it would change with time for GicV3Its. @michael2012z correct me if I am wrong?

Reply:
The state struct should be stable. But it is hard to guarantee there won't be any change in future.

There are alternative solutions:
- One is keeping GicV3ItsState in `arch` crate. But the down side is that the interface of `hypervisor` crate could be more complex for saving and reading attributes.
- Another is moving the "non-Versionized" GicV3ItsState to `hypervisor`, but adding a "Versionized" shadow struct in `arch` crate. We copy the content between them.

Reply:
If this is subjected to change then keeping a versioned structure would be a good idea, otherwise we risk breaking live migration. But if live migration is working right now, then it is highly unlikely this struct would change in future.

Reply:
If you don't use `Versionize` then you may have to do manual mangling in the future. If the hypervisor crate states are all currently then un-versioned then it seems reasonable to not version this one.

Reply:
(The `vm-migration` dependency is only required for a trivial empty `VersionMap` so you could continue to use `Versionize` and implement its own empty map which might be the approach going forward for the hypervisor crate.)

Reply:
> (The `vm-migration` dependency is only required for a trivial empty `VersionMap` so you could continue to use `Versionize` and implement its own empty map which might be the approach going forward for the hypervisor crate.)

Sorry, I didn't get your point. When I change this back to `Versionize`, the error is:
```
error[E0277]: the trait bound `Gicv3ItsState: vm_migration::VersionMapped` is not satisfied
   --> arch/src/aarch64/gic/mod.rs:605:60
    |
605 |             Snapshot::new_from_versioned_state(&self.id(), &self.state(&gicr_typers).unwrap())
    |             ----------------------------------             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `vm_migration::VersionMapped` is not implemented for `Gicv3ItsState`
    |             |
    |             required by a bound introduced by this call
    |
    = help: the following other types implement trait `vm_migration::VersionMapped`:
              MemoryRangeTable
              RegionType
note: required by a bound in `vm_migration::Snapshot::new_from_versioned_state`
   --> /home/michael/ws/src/github.com/cloud-hypervisor/cloud-hypervisor/vm-migration/src/lib.rs:194:25
    |
194 |         T: Versionize + VersionMapped,
    |                         ^^^^^^^^^^^^^ required by this bound in `vm_migration::Snapshot::new_from_versioned_state`
```

`Snapshot::new_from_versioned_state` requires something bound to `vm_migration::VersionMapped`, I didn't figure out how to workaround this.

Reply:
@rbradford Before having a better way to use `Versionize`, how about keeping the current change and continue to migrate to `hypervisor`?

Reply:
@michael2012z Sure thing!